x-pack/plugins/dashboard_enhanced/public/services/drilldowns/abstract_dashboard_drilldown/abstract_dashboard_drilldown.tsx

```diff
@@ -45,7 +45,11 @@ export abstract class AbstractDashboardDrilldown<Context extends object = object
 
   public abstract readonly supportedTriggers: () => string[];
 
-  protected abstract getLocation(config: Config, context: Context): Promise<KibanaLocation>;
+  protected abstract getLocation(
+    config: Config,
+    context: Context,
+    useUrl: boolean
```

Comment:
Nit, I wonder if we should be more explicit with the naming of this param, because even if this is false, the URL will still be _used_. What this param does is controls whether or not **app state** is placed into the URL.

Reply:
Good point! :) I made it `userUrlForState` instead - should be a lot more descriptive.